app/src/main/java/de/westnordost/streetcomplete/data/osmnotes/NotesApi.kt

```diff
@@ -0,0 +1,120 @@
+package de.westnordost.streetcomplete.data.osmnotes
+
+import de.westnordost.osmapi.OsmConnection
+import de.westnordost.osmapi.common.Handler
+import de.westnordost.osmapi.common.errors.*
+import de.westnordost.osmapi.notes.NotesDao
+import de.westnordost.osmapi.notes.QueryNotesFilters
+import de.westnordost.streetcomplete.data.osm.mapdata.BoundingBox
+import de.westnordost.streetcomplete.data.osm.mapdata.LatLon
+import de.westnordost.streetcomplete.data.osm.mapdata.toOsmApiBoundingBox
+
+
+/**
+ * Creates, comments, closes, reopens and search for notes.
+ * All interactions with this class require an OsmConnection with a logged in user.
+ *
+ * @see NotesDao which is internally used.
+ */
+class NotesApi(osm: OsmConnection) {
```

Comment:
NotesApi should be an interface. Classes that use this interface should only know the interface and don't worry about how it is constructed.

F.e. `JavaNotesApi` (or `OsmNotesApi`, `NotesApiImpl` or another name that makes sense to you) would then be the implementation of this interface for the Java/Android platform.

So, classes that need `NotesApi` to work (f.e. the NotesUploader) just declare that they need a NotesApi as constructor parameter and in the appropriate module, it is provided.

You can peek at `AndroidDatabase` on how I did this.